src/main/java/com/sap/oss/phosphor/fosstars/nvd/NVD.java

```diff
@@ -196,30 +199,26 @@ boolean downloadFailed() {
    * @throws IOException If something went wrong.
    */
   public List<NvdEntry> search(Matcher... matchers) throws IOException {
-    List<NvdEntry> result = new ArrayList<>();
+    updateIfNecessary();
 
-    for (NvdEntry entry : nvdEntries().values()) {
+    List<NvdEntry> result = new ArrayList<>();
+    Consumer<NvdEntry> collector = nvdEntry -> {
       for (Matcher matcher : matchers) {
-        if (matcher.match(entry)) {
-          result.add(entry);
+        if (matcher.match(nvdEntry)) {
+          result.add(nvdEntry);
         }
       }
+    };
+
+    if (!nvdEntries.isEmpty()) {
+      nvdEntries.forEach(collector);
```

Comment:
What is forEach doing here?

Reply:
The `forEach` applies `collector` to the NVD entries.